### PR TITLE
Use host platform for windows sdk resource compiler

### LINKED ISSUE #25870: Cross compiling windows resources to Windows Arm64 is broken
### Expected Behavior

Compiling a window resource for a binary targeting arm64 on an intel host should work.

### Current Behavior

The resource compiler ran is the one for the target, which fails to run. --info shows the resource compiler for arm64 is used. 

### Context (optional)

Add arm64 builds to our project.

### Steps to Reproduce

[GradleResourceIssue.zip](https://github.com/gradle/gradle/files/12137416/GradleResourceIssue.zip)

Attached project is just missing the wrapper for Gradle 8.2, but has everything else required. Make sure the arm64 windows compiler is installed, then run the build task.

### Gradle version

8.2

### Build scan URL (optional)

_No response_

### Your Environment (optional)

Windows 11 x64 host machine.

### PR DESCRIPTION
**Pull request by ThadHouse**
Currently, the target platform is always used for the sdk resource compiler. However, when cross compiling, this does not work, and causes resource compilation for fail. Solve for this by detecting the host platform and using that for the binary directory for the resource compiler.

Fixes #25870 

### Context
Windows Arm64 support was added as both a host and cross compiled option, however cross compiled mode does not work currently. This diff fixes that.

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [x] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [x] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [ ] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by ljacomet** [comment]
@ThadHouse Can you look into the build failures your PR causes?

**Comment by ThadHouse** [comment]
@ljacomet The failures were happening even building from master for me. Let me try rebasing to see if they're fixed.

**Comment by ThadHouse** [comment]
@ljacomet Nevermind, it was a break caused be me, just not one I understood. I added @NonNullApi to the class that one complaining, and then ran the task is told me to rerun. Is that correct?

**Comment by ljacomet** [comment]
Yes, that should be it. Let's see what the GitHub actions checks report.

**Review by ov7a — APPROVED** [review]

**Comment by ov7a** [comment]
@bot-gradle test this

**Comment by ov7a** [comment]
@ThadHouse Thank you for your contribution!

@bot-gradle test and merge

File: subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/msvcpp/DefaultWindowsSdkLocator.java [hunk 1]
```diff
@@ -37,8 +38,8 @@ public class DefaultWindowsSdkLocator implements WindowsSdkLocator {
         this.windowsKitWindowsSdkLocator = windowsKitWindowsSdkLocator;
     }
 
-    public DefaultWindowsSdkLocator(OperatingSystem operatingSystem, WindowsRegistry windowsRegistry) {
-        this(new LegacyWindowsSdkLocator(operatingSystem, windowsRegistry), new WindowsKitWindowsSdkLocator(windowsRegistry));
+    public DefaultWindowsSdkLocator(OperatingSystem operatingSystem, WindowsRegistry windowsRegistry, SystemInfo systemInfo) {
```
[the remark this answers is not in the document]
Comment (ov7a) [reply subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/msvcpp/DefaultWindowsSdkLocator.java:41]: @ThadHouse PTAL

[the remark this answers is not in the document]
Comment (ThadHouse) [reply subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/msvcpp/DefaultWindowsSdkLocator.java:41]: @ov7a Should be fixed now. Can't run those tests locally for some reason, they also fail on master on my local system, however the change was pretty simple and the fix should work.

**Comment by ov7a** [comment]
@bot-gradle test and merge

**Comment by ThadHouse** [comment]
I can't find anything in the current failures that looks related to my change.

**Comment by ov7a** [comment]
@ThadHouse Yeah, this is a performance regression, maybe CI is to blame, it had issues recently.

@bot-gradle test and merge

**Comment by ov7a** [comment]
@ThadHouse Thank you for your contribution!

**Pull request merged**